features/steps/dataset_ingress.py

```diff
@@ -38,6 +39,37 @@
         "secondary_function": dataset_ingress_v1,
     },
 }
+
+CONFIGURATION_SDMX_2_1 = {
```

Comment:
You could just add a new key to the existing configuration dictionary, e.g. `valid_no_supp_dist_v2_1` and change the other key to `valid_no_supp_dist_v2_0`. You could get rid of the separate step definitions for getting dataset IDs, in that case (on lines 102 and 107)